client/reader/discover/helper.jsx

```diff
@@ -17,11 +18,11 @@ module.exports = {
 	},
 
 	isDiscoverPost: function( post ) {
-		return !! post.discover_metadata;
+		return !! ( post.discover_metadata || post.site_ID == config( 'discover_blog_id' ) ); //eslint-disable-line eqeqeq
```

Comment:
switched to === in https://github.com/Automattic/wp-calypso/pull/4833/commits/28c011be17049738c6413d1673ab9752ad271d50